src/test/java/com/automation/test/BuyPracticeTest.java

```diff
@@ -0,0 +1,84 @@
+package com.automation.test;
+
+import com.automation.test.util.BrowserManager;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.openqa.selenium.support.ui.Wait;
+
+import java.time.Duration;
+import java.util.NoSuchElementException;
+import java.util.concurrent.TimeUnit;
+
+
+public class BuyPracticeTest {
+
+    public WebDriver driver = BrowserManager.build();
+    Wait<WebDriver> wait = new FluentWait<>(driver).withTimeout(Duration.ofSeconds(30)).ignoring(NoSuchElementException.class);
+    Actions actions = new Actions(driver);
+
+    @Test
+    public void addSomethingToCartAndBuyIt() throws InterruptedException {
+        driver.manage().timeouts().implicitlyWait(20, TimeUnit.SECONDS);
+        driver.get("http://automationpractice.com/");
+        //addToCart and Checkout
+        WebElement btnViewCloth = driver.findElement(By.xpath("//ul[@id='homefeatured']/li[1]/div"));
+
+        //Scroll to Element
+        JavascriptExecutor js = (JavascriptExecutor) driver;
+        js.executeScript("arguments[0].scrollIntoView(true);", btnViewCloth);
+        wait.until(ExpectedConditions.visibilityOf(btnViewCloth));
+        actions.moveToElement(btnViewCloth).perform();
```

Comment:
why do you need to do scroll

Reply:
That's scroll it's needed to run the automation with firefox

Reply:
use scroll into your tests add flakiness, please try to find a solution without scroll